packages/block-library/src/gallery/v1/edit.js

```diff
@@ -220,9 +220,9 @@ function GalleryEdit( props ) {
 			return currentImageCaption;
 		}
 
-		const attachment = find( attachmentCaptions, {
-			id: newImageId,
-		} );
+		const attachment = attachmentCaptions.find(
```

Comment:
Will always be an array or `undefined`:

https://github.com/WordPress/gutenberg/blob/7d10ef1fe6d69c9844f7ca3b82ed5d799adbebfc/packages/block-library/src/gallery/v1/edit.js#L237

However, a few lines above (L220) we check for the falsy scenario, so it will always be an array:

https://github.com/WordPress/gutenberg/blob/7d10ef1fe6d69c9844f7ca3b82ed5d799adbebfc/packages/block-library/src/gallery/v1/edit.js#L219-L222